code/datums/keybinding/artificial_intelligence.dm

```diff
@@ -0,0 +1,21 @@
+/datum/keybinding/artificial_intelligence
+	category = CATEGORY_AI
+	weight = WEIGHT_AI
+
+/datum/keybinding/artificial_intelligence/can_use(client/user)
+	return isAI(user.mob)
+
+/datum/keybinding/artificial_intelligence/reconnect
+	hotkey_keys = list("-")
+	name = "reconnect"
+	full_name = "Reconnect to shell"
+	description = "Reconnects you to your most recently used AI shell"
+	keybind_signal = COMSIG_KB_SILICON_RECONNECT_DOWN
+
+/datum/keybinding/artificial_intelligence/reconnect/down(client/user)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/silicon/ai/our_ai = user.mob
+	our_ai.deploy_to_shell(our_ai.redeploy_action.last_used_shell)
```

Comment:
What happens when this is null or the shell was destroyed?

Reply:
This is a good question, I'll add a check to make sure last_used_shell isn't null before it is passed through to deploy_to_shell.

Reply:
deploy_to_shell handles null / destroyed / dead shells by printing to chat "No usable AI shell beacons detected." Which is more feedback than my early return provided, so it's probably better to let it handle itself.